### DISCUSSION TITLE
Unsupported field: constructors

### DISCUSSION THREAD

**Discussion by enter-marlah** [apache/camel-karavan#1154] [Q&A] [answered]
Hello!

I'm trying to build a custom kamelet but when I add a bean to the kamelet and run the project, I get an error Unsupported field: constructors.

The generated yaml looks like this:
    beans:
      - constructors: {}
        name: testBean
        type: java.util.HashMap

If I do the same in camel route it works fine.

What am I doing wrong?

Thanks!

**Comment by mgubaidullin** [comment]
@davsclaus should beans in Kamelets behave the same as beans in plain yaml DSL and support `constructors`?

**Comment by davsclaus** [comment]
Constructors to beans in yaml dsl is rather new. Are you perhaps using Camel K by any chance as it tend to be using an older Camel version. Also Karavan 4.4 is on the way that comes with latest Camel 4.4.0 release, you can try when its released soon.

**Reply by enter-marlah** [reply]
I'm quite new to Camel in general. I installed the Karavan into my kubernetes cluster from the main branch from the Docs found here https://github.com/apache/camel-karavan/tree/main/docs/install/karavan-kubernetes

The image in the deployment.yaml is ghcr.io/apache/camel-karavan:4.4.0

Do I need to change something to get the constructor working in the Kamelet or is that something that Kamelets dont support at all?

**Reply by davsclaus** [reply]
Do you have a sample kamelet etc that you can show us, so we can use that to look into this.

**Comment by davsclaus** [comment, marked as answer]
Okay constructors is not supported in Kamelets at this time

**Reply by davsclaus** [reply]
I created a ticket: https://issues.apache.org/jira/browse/CAMEL-20514

**Reply by enter-marlah** [reply]
Thanks for your help!
